beacon-chain/lightclient/config.go

```diff
@@ -36,3 +58,54 @@ func NewConfig(chainConfig *params.BeaconChainConfig) *Config {
 		SecondsPerSlot:               chainConfig.SecondsPerSlot,
 	}
 }
+
+func NewConfigFromJSON(config *ConfigJSON) (*Config, error) {
```

Comment:
This is fine, but maybe we can use `json.Unmarshaler` interface?  (Or added a wrapper to call NewConfigFromJSON)
 
 So we can use standard json interface (and will be good for nested decoding automatically)

Reply:
mm how would unmarshaler help here? We're producing a different kind of object

Reply:
are you suggesting we marshal to ConfigJSON -> bytes and unmarshal the other direction?

Reply:
check my next commit let me know if that's what you meant